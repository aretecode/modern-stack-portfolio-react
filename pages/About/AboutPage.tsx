/**
 * @see https://material.io/design/components/cards.html#actions
 * @file @todo move out the `⇔` @@hack
 * @todo on mobile, could expand to full screen
 */
import * as React from 'react'
import { PortfolioHead } from '../../src/features/PortfolioHead'
import { PortfolioContext } from '../../src/features/PortfolioContext'
import { PortfolioSchema } from '../../src/features/PortfolioSchema'
import { Address } from './Address'
import {
  StyledSocialProfiles,
  StyledName,
  StyledSummary,
  StyledLabel,
  StyledAboutMeArticle,
  StyledAboutMeImg,
  StyledArrow,
  StyledLink,
  StyledContactNav,
  StyledFigCaption,
  StyledFigure,
  StyledTextLineSeparator,
  StyledCardDivider,
} from './styled'
import { Skills } from './Skills'
import {
  AnimateHeightContextProvider,
  AnimateHeightContext,
} from '../../src/features/AnimateHeight'

/**
 * @see https://reactjs.org/docs/hooks-reference.html#usecontext
 */
function AboutMeImage(props: React.ComponentProps<typeof StyledAboutMeImg>) {
  const value = React.useContext(AnimateHeightContext)
  return <StyledAboutMeImg {...props} isExpanded={value.isExpanded} />
}
function CardDivider(props: {}) {
  const value = React.useContext(AnimateHeightContext)
  return <StyledCardDivider isExpanded={value.isExpanded} />
}

export function AboutPage() {
  const {
    name,
    label = '',
    picture,
    summary,
    telephone,
    email,
  } = React.useContext(PortfolioContext).basics

  const titleText = `About ${name}`
  return (
    <>
      <PortfolioHead titleText={titleText} description={summary} />
      <PortfolioSchema />
      <AnimateHeightContextProvider>
        <StyledAboutMeArticle>
          <StyledSocialProfiles />
          <StyledFigure>
            <AboutMeImage
              src={picture}
              height={'600'}
              width={'600'}
              alt="about me picture"
            />
            <StyledFigCaption>
              <StyledName>{name}</StyledName>
              <StyledLabel>
                {label.split('⇔').shift()}
                <StyledArrow>↔</StyledArrow>
                {label.split('⇔').pop()}
              </StyledLabel>
              <StyledTextLineSeparator />
              <StyledSummary>{summary}</StyledSummary>
              <StyledContactNav>
                <section>
                  <header>Phone</header>
                  <StyledLink to={`tel:${telephone}`}>+{telephone}</StyledLink>
                </section>
                <section>
                  <header>Email</header>
                  <StyledLink to={`mailto:${email}`}>{email}</StyledLink>
                </section>
              </StyledContactNav>
            </StyledFigCaption>
          </StyledFigure>
          <CardDivider />
          <Skills />
        </StyledAboutMeArticle>
      </AnimateHeightContextProvider>
    </>
  )
}
