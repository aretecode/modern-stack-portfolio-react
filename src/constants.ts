export const defaultApolloStatePortfolio = {
  __typename: 'Resume',
  basics: {
    __typename: 'Basics',
    name: '',
    label: '',
    picture: '',
    email: 'james@jameswiens.com',
    telephone: '12506509455',
    website: '',
    summary: '',
    profiles: [
      {
        __typename: 'Profile',
        network: 'linkedin',
        username: 'aretecode',
        url: 'https://www.linkedin.com/in/james-wiens/',
      },
    ],
    address: '',
    postalCode: '',
    city: '',
    countryCode: '',
    region: '',
    resumeWebsite: '',
    skills: ['skill1'],
  },
  work: [
    {
      __typename: 'Work',
      company: 'Open Source',
      position: '',
      website: 'https://github.com/aretecode',
      startDate: '01/02/2013',
      endDate: 'current',
      summary: '',
      highlights: '',
      picture:
        'https://user-images.githubusercontent.com/4022631/55686780-04f0f980-5983-11e9-8152-204681b0840f.png',
    },
  ],
} as const
