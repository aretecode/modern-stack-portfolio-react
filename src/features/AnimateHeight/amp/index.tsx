/**
 * Amp AnimateHeight
 */
import * as React from 'react'
import Head from 'next/head'
import { defaultRenderTrigger } from '../renderProps'
import { AmpAnimateHeightProps } from '../typings'

export function AmpAnimateHeight(props: AmpAnimateHeightProps) {
  const { renderTrigger = defaultRenderTrigger, children } = props

  /**
   * @see https://amp.dev/documentation/components/amp-accordion?referrer=ampproject.org
   * @see /packages/client/pages/About/AmpStyles.tsx
   * @todo split to a component
   */
  return (
    <>
      <Head>
        <script
          async
          custom-element="amp-accordion"
          src="https://cdn.ampproject.org/v0/amp-accordion-0.1.js"
        />
      </Head>
      <amp-accordion layout="container">
        <section>
          <header className="amp-accordion-header">
            {renderTrigger({ isExpanded: false } as any)}
          </header>
          {children}
        </section>
      </amp-accordion>
    </>
  )
}
