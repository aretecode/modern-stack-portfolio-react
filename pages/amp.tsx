/**
 * @see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34801
 */
// import { withAmp } from 'next/amp'
import AboutPage from '../src/pages/About/amp/AmpAboutPage'

// export default withAmp(AboutPage)
export default AboutPage
