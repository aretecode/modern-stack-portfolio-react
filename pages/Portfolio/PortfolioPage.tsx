// tslint:disable
import * as React from 'react'
import {
  PortfolioContext,
  PortfolioContextType,
} from '../../src/features/PortfolioContext'
import { PortfolioHead } from '../../src/features/PortfolioHead'
import { StyledCard } from '../../src/features/Card'
import { StyledMain } from '../../src/features/Main'
import { StyledLink } from '../../src/features/Link'
import { WorkType } from '../../src/typings'
import {
  StyledGrid,
  StyledCardImage,
  StyledCardFigure,
  StyledLeaderBoard,
} from './styled'
import { TimeRange } from './TimeRange'

/**
 * @hack images with specific sizes
 */
function renderWork(work: WorkType, index: number) {
  return (
    <StyledCard key={work.company + work.startDate + work.endDate}>
      <StyledCardFigure>
        <StyledCardImage
          width="1000"
          height="692"
          isAlwaysAboveTheFold={index === 0}
          src={work.picture}
          loading="lazy"
          srcSizeList={[
            ['(max-width: 800px)', work.picture.replace('m-', 'm-')],
            ['(max-width: 1024px)', work.picture.replace('m-', 'w-m-')],
            ['(min-width: 2000px)', work.picture.replace('m-', 'xl-')],
          ]}
          alt={`work picture for ${work.company}`}
        />
        <figcaption>
          <header>{work.company}</header>
          <section>
            <strong>{work.position}</strong>
            {typeof work.highlights === 'string' &&
            work.highlights.includes('- ') ? (
              <ul key="highlights">
                {work.highlights
                  .split('- ')
                  .filter(Boolean)
                  .map(highlight => (
                    <li key={highlight}>{highlight}</li>
                  ))}
              </ul>
            ) : (
              <p key="highlights">{work.highlights}</p>
            )}
            <p>{work.summary}</p>
            <StyledLink to={work.website}>{work.website}</StyledLink>
          </section>
          <section>
            <TimeRange startDate={work.startDate} endDate={work.endDate} />
          </section>
        </figcaption>
      </StyledCardFigure>
    </StyledCard>
  )
}

/**
 * could provide a cool graph to sort resumes/portfolios too and highlight
 * like build your own github
 */
export class PortfolioPage extends React.PureComponent {
  static contextType = PortfolioContext
  readonly context: PortfolioContextType

  render() {
    const { name, summary } = this.context.basics
    const titleText = `${name}'s Portfolio`

    /**
     * @todo move static url out
     */
    const imageUrl =
      'https://noccumpr-cdn.sirv.com/images/james-wiens-work-experience-combined-filtered.png'

    return (
      <>
        <PortfolioHead
          titleText={titleText}
          description={summary}
          image={imageUrl}
        />
        <StyledMain>
          <StyledLeaderBoard>
            <h1>What I've done</h1>
          </StyledLeaderBoard>
          <StyledGrid>{this.context.work.map(renderWork)}</StyledGrid>
        </StyledMain>
      </>
    )
  }
}
