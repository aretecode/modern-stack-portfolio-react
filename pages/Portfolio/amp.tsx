/**
 * @see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34801
 */
import { withAmp } from 'next/amp'
import PortfolioPage from '../../src/pages/Portfolio/PortfolioPage'

export default withAmp(PortfolioPage)
