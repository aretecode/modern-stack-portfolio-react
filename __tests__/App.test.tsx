/**
 * @see https://github.com/zeit/next.js/blob/master/examples/with-jest/__tests__/index.test.js
 */
import * as React from 'react'
import * as useAmpModule from 'next/amp'
import { waitFor } from '@testing-library/react'
import { render } from './render'
import { defaultApolloStatePortfolio } from '../src/constants'
import AboutPage from '../pages/About'
import PortfolioPage from '../pages/Portfolio'

const sleep = (time: number) => Promise.resolve(time)

describe('<App>', () => {
  it('should match snapshot', async () => {
    const view = (
      <>
        <PortfolioPage />
        <AboutPage />
      </>
    )
    const { container, rerender, getByText } = render(view)
    rerender(view)
    console.log('[tests] App - waiting')
    await waitFor(() =>
      expect(() =>
        getByText(defaultApolloStatePortfolio.work[0].company)
      ).not.toThrowError()
    )
    await sleep(5000)

    console.log('[tests] App - waited')
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot for amp', async () => {
    jest.spyOn(useAmpModule, 'useAmp').mockImplementation(() => true)

    const view = (
      <>
        <PortfolioPage />
        <AboutPage />
      </>
    )
    const { container, rerender, getByText } = render(view)
    rerender(view)
    console.log('[tests] App - waiting')
    await waitFor(() =>
      expect(() =>
        getByText(defaultApolloStatePortfolio.work[0].company)
      ).not.toThrowError()
    )
    await sleep(5000)

    console.log('[tests] App - waited')
    expect(container).toMatchSnapshot()
  })
})
