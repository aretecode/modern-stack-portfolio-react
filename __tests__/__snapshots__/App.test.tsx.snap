// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should match snapshot 1`] = `
.c5 {
  text-decoration: none;
  position: relative;
  letter-spacing: 0.2em;
  color: var(--color-link);
}

.c5:link,
.c5:visited {
  color: var(--color-link);
}

.c5:focus {
  outline: thin dotted;
}

.c5:hover {
  color: #aaa;
}

.c5:link {
  -webkit-tap-highlight-color: rgba(102,102,102,0.5);
}

.c34 {
  background-color: var(--color-material-background-purple);
  padding: 2rem 1.25rem;
}

.c34 p {
  color: var(--color-text-body);
}

.c35 {
  color: #df0909;
  display: inline-flex;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  height: 12px;
}

.c36 {
  padding-right: 0.5rem;
  padding-left: 0.5rem;
  transform: translateY(1px);
  display: inline-flex;
  width: 38px;
  height: 14px;
}

.c37 {
  text-decoration: none;
  position: relative;
  letter-spacing: 0.2em;
  color: var(--color-link);
  padding-left: 0.3rem;
  text-decoration: underline;
}

.c37:link,
.c37:visited {
  color: var(--color-link);
}

.c37:focus {
  outline: thin dotted;
}

.c37:hover {
  color: #aaa;
}

.c37:link {
  -webkit-tap-highlight-color: rgba(102,102,102,0.5);
}

.c4 {
  display: flex;
  justify-content: space-around;
  padding: 0;
  margin: 0;
}

.c4 > li {
  display: inline-flex;
}

.c4 > li > a {
  padding: 1rem;
}

.c0 {
  background-color: var(--color-material-background-purple);
  position: absolute;
  background: rgba(0,0,0,0.2);
  top: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;
  width: 100%;
  height: 4rem;
  padding-left: 1.5rem;
  padding-right: 0.5rem;
}

.c2 {
  color: var(--color-text-body);
}

.c1 {
  text-decoration: none;
  position: relative;
  letter-spacing: 0.2em;
  color: var(--color-link);
}

.c1:link,
.c1:visited {
  color: var(--color-link);
}

.c1:focus {
  outline: thin dotted;
}

.c1:hover {
  color: #aaa;
}

.c1:link {
  -webkit-tap-highlight-color: rgba(102,102,102,0.5);
}

.c38 {
  height: 0;
  width: 0;
  visibility: hidden;
  position: absolute;
}

.c14 {
  padding: 1rem;
  width: 75px;
  display: inline-flex;
  justify-content: space-around;
  flex-direction: column;
  align-items: center;
  object-fit: contain;
  object-position: bottom;
  transition: flex-direction 0.24s ease-in-out,width 0.24s ease-in-out;
}

.c15 {
  text-decoration: none;
  position: relative;
  letter-spacing: 0.2em;
  color: var(--color-link);
}

.c15:link,
.c15:visited {
  color: var(--color-link);
}

.c15:focus {
  outline: thin dotted;
}

.c15:hover {
  color: #aaa;
}

.c15:link {
  -webkit-tap-highlight-color: rgba(102,102,102,0.5);
}

.c16 {
  width: 2rem;
  height: 2rem;
}

.c12 {
  color: var(--color-text-body);
  display: inline-flex;
  padding: 0 0.5rem;
}

.c26 {
  text-decoration: none;
  position: relative;
  letter-spacing: 0.2em;
  color: var(--color-link);
}

.c26:link,
.c26:visited {
  color: var(--color-link);
}

.c26:focus {
  outline: thin dotted;
}

.c26:hover {
  color: #aaa;
}

.c26:link {
  -webkit-tap-highlight-color: rgba(102,102,102,0.5);
}

.c26.c26.c26 {
  color: unset;
  text-decoration: initial;
  letter-spacing: initial;
}

.c27 {
  color: var(--color-text-body);
  display: inline-flex;
  padding: 0 0.5rem;
  width: 90%;
  border: none;
  border-top: 1px solid var(--color-dark-background-light);
  border-bottom: 1px solid var(--color-dark-background-dark);
  opacity: 0.1;
  margin: 0 auto;
  margin-top: 1rem;
  padding: 0;
  height: 2px;
  order: 1;
  transition: margin-top 0.96s ease-in-out,margin-left 0.24s ease-in-out;
}

.c23 {
  color: var(--color-text-body);
  display: inline-flex;
  padding: 0 0.5rem;
  color: var(--color-text-secondary);
  border: none;
  border-top: 5px solid var(--color-text-secondary);
  margin: 0;
  padding: 0;
  height: 0.75rem;
  width: 8vw;
}

.c20 {
  font-size: 2rem;
  margin-bottom: 0;
}

.c22 {
  font-style: unset;
  padding-right: 0.15rem;
  opacity: 0.3;
}

.c24 {
  margin-top: 1rem;
}

.c21 {
  color: var(--color-dark-background-main);
  margin: 0;
  margin-top: -0.5rem;
  font-size: 1.2rem;
}

.c18 {
  display: flex;
  object-fit: cover;
  border-radius: 0.125rem;
  box-shadow: none;
  transition: margin-top 0.5s ease-in-out,max-width 0.24s ease-in-out, height 1s ease-in-out,object-position 0.8s ease-in-out, box-shadow 0.24s ease-in-out;
}

.c17 {
  margin: 0;
  padding: 0;
  display: flex;
  width: calc(100% - 75px);
}

.c19 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  word-break: break-word;
}

.c25 {
  display: flex;
  padding: 2rem 0;
  transition: flex-direction 500ms ease-in-out;
}

.c25 > section {
  flex-basis: 50%;
}

.c25 > section > header {
  font-weight: bold;
}

.c13 {
  background-color: #fff;
  margin: 9rem 1rem 9rem 1rem;
  border-radius: 0.125rem;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.2), 0 1px 5px 0 rgba(0,0,0,0.12);
  display: flex;
  flex-wrap: wrap;
  width: 80%;
}

.c13 address {
  color: var(--color-text-unimportant);
}

.c30 {
  width: 42px;
  height: 42px;
}

.c28 {
  width: 100%;
  display: flex;
  justify-content: flex-end;
  transition: justify-content 500ms ease-in-out;
  margin-right: 1rem;
}

.c29 {
  background: none;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  padding: 0;
  appearance: none;
  user-select: none;
}

.c31 {
  overflow: hidden;
  transition: margin-top 0.62s ease-in-out,height 0.24s ease-out;
  height: 0;
  background: white;
  width: 100vw;
  flex-basis: 100%;
  display: flex;
  flex-wrap: wrap;
  order: 4;
}

.c33 {
  display: inline-flex;
  border-radius: 4px;
  margin: 0.25rem;
  padding: 0.5rem;
  background: var(--color-dark-background-dark);
  color: white;
  border-radius: 6px;
}

.c32 {
  margin: 0;
  padding: 1rem;
}

.c8 {
  color: var(--color-text-body);
  background-color: var(--color-dark-background-main);
  margin: 0;
  padding: 0;
  position: relative;
  min-width: 0;
  word-wrap: break-word;
  word-break: break-word;
  overflow-wrap: break-word;
  white-space: normal;
  background-clip: border-box;
  border-radius: 0.125rem;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 3px 1px -2px rgba(0,0,0,0.2), 0 1px 5px 0 rgba(0,0,0,0.12);
}

.c8 figcaption {
  padding: 1.25rem;
}

.c8 header {
  color: var(--color-text-secondary);
  font-size: 1.5rem;
  font-weight: 300;
}

.c8 div[role='separator'],
.c8 time {
  font-size: 80%;
  font-weight: 400;
  color: var(--color-text-unimportant);
}

.c8 a {
  color: var(--color-text-unimportant);
  border-color: var(--color-text-unimportant);
  border-width: 1px;
}

.c8 a:link,
.c8 a:visited {
  color: var(--color-text-unimportant);
}

.c8 a::after {
  display: none;
}

.c7 {
  display: grid;
  padding: 1rem;
  grid-gap: 0.5rem;
  grid-auto-rows: auto;
}

.c11 {
  display: inline-flex;
}

.c10 {
  filter: url(#green-tint);
  max-width: 100%;
}

.c10 > img {
  max-width: 100%;
}

.c9 {
  margin: 0;
  padding: 0;
}

.c6 {
  background-color: var(--color-orange);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 4rem 0 2rem 0;
}

.c6 > h1 {
  color: white;
  font-size: 3em;
  font-weight: 300;
  text-shadow: 0 1px 0 rgba(255,255,255,0.55),0 2px 0 rgba(0,0,0,0.1), 0 3px 0 rgba(0,0,0,0.1),0 4px 0 rgba(0,0,0,0.1);
}

@media (max-width:480px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:480px) {
  .c0 {
    height: 7rem;
  }
}

@media (max-width:480px) {
  .c2 {
    text-align: center;
    margin-bottom: 0;
  }
}

@media (max-width:480px) {
  .c1 {
    width: 100%;
    flex-basis: 100%;
  }
}

@media (max-width:1023px) {
  .c14 {
    width: 75%;
    flex-direction: row;
    order: 2;
    height: unset;
    justify-content: unset;
    padding: 0 0 0.5rem 0;
  }
}

@media (max-width:1023px) {
  .c15 {
    flex-basis: 33%;
    width: 33%;
    text-align: center;
  }
}

@media (max-width:1023px) {
  .c27 {
    margin-bottom: 1rem;
  }
}

@media (min-width:1024px) {
  .c27 {
    visibility: hidden;
    margin: 0;
  }
}

@media (min-width:1024px) {
  .c20 {
    margin-top: 3rem;
  }
}

@media (max-width:1024px) {
  .c20 {
    flex-basis: 100%;
  }
}

@media (min-width:1024px) {
  .c24 {
    flex-basis: 100%;
  }
}

@media (max-width:1024px) {
  .c24 {
    flex-basis: 100%;
  }
}

@media (max-width:1024px) {
  .c21 {
    flex-basis: 100%;
  }
}

@media (max-width:500px) {
  .c18 {
    object-position: left;
  }
}

@media (max-width:1023px) {
  .c18 {
    width: 100%;
  }
}

@media (min-width:1024px) {
  .c18 {
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.2),0 1px 5px 0 rgba(0,0,0,0.12);
    height: 120%;
    object-position: left;
    margin-top: -3rem;
    max-width: 21rem;
    flex-basis: 30%;
  }
}

@media (max-width:1023px) {
  .c17 {
    flex-wrap: wrap;
    width: 100%;
  }
}

@media (max-width:1023px) {
  .c19 {
    display: flex;
    flex-wrap: wrap;
  }
}

@media (max-width:1023px) {
  .c25 {
    padding: 0.5rem 0;
    width: 100%;
  }
}

@media (max-width:1023px) {
  .c13 {
    width: calc(100% - 2rem);
    height: unset;
    flex-wrap: wrap;
    margin: 5.5rem 1rem 2rem 1rem;
  }
}

@media (max-width:480px) {
  .c13 {
    margin: 8rem 1rem 2rem;
  }
}

@media (max-width:1023px) {
  .c28 {
    justify-content: center;
    flex-basis: 25%;
    order: 3;
    margin-right: 0;
    margin-top: -0.25rem;
  }
}

@media (min-width:1024px) {
  .c28 {
    margin-top: -3.5rem;
  }
}

@media (max-width:1023px) {
  .c29 {
    width: 42px;
    height: 42px;
  }
}

@media (min-width:1024px) {
  .c8 {
    padding: 0 1rem 0 0;
  }
}

@media (max-width:1023px) {
  .c8 {
    box-shadow: 0 1rem 0.75rem rgba(0,0,0,0.19), 0 0.5rem 0.5rem rgba(0,0,0,0.23);
  }
}

@media (min-width:1024px) {
  .c9 {
    display: grid;
    grid-template-columns: calc(50vw - 2rem) calc(50vw - 2rem);
    grid-gap: 1rem;
  }

  .c9 picture,
  .c9 img {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  .c9 figcaption {
    padding: 1.25rem 0;
  }
}

@media (max-width:1023px) {
  .c9 img {
    max-height: 60vh;
  }
}

@media (max-width:480px) {
  .c6 {
    padding-top: 8rem;
  }
}

<div>
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@graph":[{"@type":"Person","name":"","sameAs":["https://www.linkedin.com/in/james-wiens/"],"url":"","email":"james@jameswiens.com","telephone":"12506509455","address":{"@type":"PostalAddress","addressLocality":"","addressRegion":"","postalCode":"","streetAddress":""}},{"@type":"Organization","name":"Open Source","url":"?company=Open Source","member":{"@type":"OrganizationRole","member":{"@type":"Person","name":"","sameAs":["https://www.linkedin.com/in/james-wiens/"],"url":"","email":"james@jameswiens.com","telephone":"12506509455","address":{"@type":"PostalAddress","addressLocality":"","addressRegion":"","postalCode":"","streetAddress":""}},"startDate":"01/02/2013","endDate":"current","roleName":""}},{"@type":"ItemList","itemListElement":[{"@type":"ListItem","position":1,"item":{"@type":"Organization","name":"Open Source","url":"?company=Open Source","member":{"@type":"OrganizationRole","member":{"@type":"Person","name":"","sameAs":["https://www.linkedin.com/in/james-wiens/"],"url":"","email":"james@jameswiens.com","telephone":"12506509455","address":{"@type":"PostalAddress","addressLocality":"","addressRegion":"","postalCode":"","streetAddress":""}},"startDate":"01/02/2013","endDate":"current","roleName":""}}}]}]}
  </script>
  <header
    class="c0"
    role="banner"
  >
    <a
      class="c1 "
      href="/"
    >
      <p
        class="c2"
        id="logo"
      />
    </a>
    <nav
      class="c3"
    >
      <ul
        class="c4"
      >
        <li>
          <a
            class="c5 "
            href="/Portfolio"
            rel="me"
          >
            work
          </a>
        </li>
        <li>
          <a
            class="c5 "
            href="/"
            rel="me"
          >
            about
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class=""
  >
    <header
      class="c6"
    >
      <h1>
        What I've done
      </h1>
    </header>
    <div
      class="c7"
    >
      <article
        class="c8"
      >
        <figure
          class="c9"
        >
          <picture
            class="c10"
          >
            <source
              media="(max-width: 800px)"
              srcset="https://user-images.githubusercontent.com/4022631/55686780-04f0f980-5983-11e9-8152-204681b0840f.png"
            />
            <source
              media="(max-width: 1024px)"
              srcset="https://user-images.githubusercontent.com/4022631/55686780-04f0f980-5983-11e9-8152-204681b0840f.png"
            />
            <source
              media="(min-width: 2000px)"
              srcset="https://user-images.githubusercontent.com/4022631/55686780-04f0f980-5983-11e9-8152-204681b0840f.png"
            />
            <img
              alt="work picture for Open Source"
              class=""
              loading="lazy"
              src="https://user-images.githubusercontent.com/4022631/55686780-04f0f980-5983-11e9-8152-204681b0840f.png"
            />
            <noscript />
          </picture>
          <figcaption>
            <header>
              Open Source
            </header>
            <section>
              <strong />
              <p />
              <p />
              <a
                class="c5"
                href="https://github.com/aretecode"
              >
                https://github.com/aretecode
              </a>
            </section>
            <section>
              <time
                class="c11"
              >
                01/02/2013
              </time>
              <div
                class="c12"
                role="separator"
              >
                -
              </div>
              <time
                class="c11"
              >
                current
              </time>
            </section>
          </figcaption>
        </figure>
      </article>
    </div>
  </main>
  <article
    class="c13"
  >
    <aside
      class="c14"
    >
      <a
        class="c15"
        href="https://www.linkedin.com/in/james-wiens/"
        rel="me"
        title="aretecode"
      >
        <svg
          class="c16"
          role="img"
          viewBox="0 0 24 24"
          xmlns="https://www.w3.org/2000/svg"
        >
          <title>
            linkedin
             icon
          </title>
          <path
            d="M16.893 20.452v-5.569c0-1.328-.027-3.037-1.852-3.037-1.853 0-2.136 1.445-2.136 2.939v5.667H9.351V9h3.414v1.561h.046c.477-.9 1.637-1.85 3.37-1.85 3.601 0 4.267 2.37 4.267 5.455v6.286h-3.555zM5.337 7.433a2.062 2.062 0 0 1-2.063-2.065 2.064 2.064 0 1 1 2.063 2.065zm1.782 13.019H3.555V9h3.564v11.452z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </a>
    </aside>
    <figure
      class="c17"
    >
      <img
        alt="about me picture"
        class="c18 "
        src=""
      />
      <figcaption
        class="c19"
      >
        <h3
          class="c20"
        />
        <h4
          class="c21"
        >
          
          <i
            class="c22"
          >
            ↔
          </i>
          
        </h4>
        <hr
          class="c23"
          role="separator"
        />
        <p
          class="c24"
        />
        <nav
          class="c25"
        >
          <section>
            <header>
              Phone
            </header>
            <a
              class="c26 "
              href="tel:12506509455"
            >
              +
              12506509455
            </a>
          </section>
          <section>
            <header>
              Email
            </header>
            <a
              class="c26 "
              href="mailto:james@jameswiens.com"
            >
              james@jameswiens.com
            </a>
          </section>
        </nav>
      </figcaption>
    </figure>
    <hr
      class="c27"
      role="separator"
    />
    <div
      class="c28"
      title="show/hide skills"
    >
      <button
        class="c29"
      >
        <svg
          class="c30"
          role="img"
          title="show"
          viewBox="0 0 24 24"
          xmlns="https://www.w3.org/2000/svg"
        >
          <title>
            down_arrow
             icon
          </title>
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </button>
    </div>
    <aside
      aria-hidden="true"
      class="c31"
    >
      <ul
        class="c32"
      >
        <li
          class="c33"
        >
          skill1
        </li>
      </ul>
    </aside>
  </article>
  <footer
    class="c34"
  >
    <p>
      <span>
        ©
        2019
      </span>
       
      <span />
    </p>
    <p>
      <span
        class=""
      >
        Made with
      </span>
      <svg
        class="c35"
        viewBox="0 0 54 50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Heart
        </title>
        <path
          d="M24.873 47.201c-1.162-1.341-4.064-3.956-6.449-5.81-7.066-5.496-8.028-6.291-10.897-9.012C2.238 27.364-.008 22.325 0 15.493c.004-3.336.224-4.62 1.131-6.587 1.54-3.335 3.806-5.814 6.704-7.33C9.887.504 10.899.027 14.327.007c3.585-.02 4.34.41 6.449 1.604 2.567 1.452 5.21 4.556 5.756 6.76l.337 1.363.832-1.876c4.7-10.598 19.71-10.44 24.933.264 1.657 3.395 1.839 10.645.37 14.729-1.917 5.327-5.517 9.389-13.839 15.612-5.457 4.082-11.634 10.258-12.064 11.125-.499 1.007-.024.158-2.228-2.386z"
          fill="red"
          fill-rule="nonzero"
        />
      </svg>
      <span
        class=""
      >
        in
      </span>
      <svg
        class="c36"
        viewBox="0 0 50 25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Canada, eh
        </title>
        <g
          fill="none"
          fill-rule="nonzero"
        >
          <path
            d="M0 0h50v25H0z"
            fill="red"
          />
          <path
            d="M12.5 0h25v25h-25z"
            fill="#FFF"
          />
          <path
            d="M25 2.344l-1.706 3.181c-.194.346-.54.314-.887.12l-1.235-.639.92 4.886c.194.893-.427.893-.733.507l-2.156-2.412-.35 1.225c-.04.16-.217.33-.483.29l-2.725-.574.715 2.603c.154.579.273.819-.154.971l-.972.457 4.691 3.81c.186.144.28.404.214.638l-.41 1.348c1.614-.187 3.062-.467 4.678-.64.142-.014.381.221.38.386l-.214 4.936h.786l-.124-4.925c-.001-.165.216-.411.358-.396 1.616.172 3.064.452 4.679.639l-.41-1.348a.614.614 0 0 1 .213-.638l4.69-3.81-.97-.457c-.428-.152-.309-.392-.155-.971l.715-2.603-2.725.573c-.266.04-.443-.128-.483-.29l-.35-1.224-2.155 2.412c-.307.386-.928.386-.734-.506l.92-4.887-1.235.64c-.347.193-.693.225-.887-.121"
            fill="red"
          />
        </g>
      </svg>
    </p>
    <p>
      <span
        class=""
      >
        Open Sourced at
      </span>
      <a
        class="c37"
        href="https://github.com/aretecode/modern-stack-web-portfolio"
      >
        github.com/aretecode
      </a>
    </p>
  </footer>
  <svg
    class="c38"
  >
    <filter
      color-interpolation-filters="sRGB"
      height="100%"
      id="green-tint"
      width="100%"
      x="0"
      y="0"
    >
      <fecolormatrix
        type="matrix"
        values="0.36 0 0 0  0.02
              0.56 0 0 0  0.06
              0.40 0 0 0  0.16
                0  0 0 1  0"
      />
    </filter>
  </svg>
</div>
`;
